metricbeat/module/docker/docker.go

```diff
@@ -136,7 +137,18 @@ func exportContainerStats(ctx context.Context, client *client.Client, container
 	var event Stat
 	event.Container = container
 
-	containerStats, err := client.ContainerStats(ctx, container.ID, false)
+	// stream defaults to false -
+	// force container stats stream to receive also precpu_stats when using podman
+	// docker returns precpu_stats with or without streaming, podman does only return precpu_stats when streaming
+	// this can be done using the label co.elastic.metricbeat.docker.forceContainerStatsStream
+	var stream bool
+	if container.Labels != nil {
+		if labelValue, ok := container.Labels["co.elastic.metricbeat.docker.forceContainerStatsStream"]; ok {
+			stream, _ = strconv.ParseBool(labelValue)
```

Comment:
If you set the stream to true , what is the effect on the response time? 
Trying it locally the `stream=1` parameter in the http client never returns.
```
curl -s --unix-socket /var/run/docker.sock -H 'Content-Type: application/json' http://localhost/containers/<contId>/stats\?stream\=1
```
It is like using `docker stats` vs `docker stats --no-stream`.
When stream=false the docker engine fetches 2 stats for the container to be able to calculate percentages. See https://github.com/moby/moby/issues/40437.
In general the best approach would be to use the newer https://github.com/moby/moby/blob/f472dda2e9dec18d4bf011fa6b2c5c388d6a20a7/client/container_stats.go#L32 as the percentages are calculated at the collector side (metricbeat).

Regarding the Podman error in fetching memory stats, we will investigate it to understand why this happens

Reply:
FYI until now I have made a local image from this branch code.
I have installed podman in my MacOs and created a local container 

I have added the label:
```bash
podman container inspect 45394fdd9ab9

"Labels": {
                    "co.elastic.metricbeat.docker.forceContainerStatsStream": "true",
                    "maintainer": "NGINX Docker Maintainers <docker-maint@nginx.com>"
```

Still error persists :
```bash
{"log.level":"error","@timestamp":"2024-01-31T16:43:04.321+0200","log.origin":{"function":"github.com/elastic/beats/v7/metricbeat/mb/module.(*metricSetWrapper).fetch","file.name":"module/wrapper.go","file.line":256},"message":"Error fetching data for metricset docker.memory: No memory stats data available","service.name":"metricbeat","ecs.version":"1.6.0"}
```